### PR TITLE
Holy water makes you magic-immune / adds holy water OD

### PR DESCRIPTION
**Pull request by gorillion**
## About The Pull Request

Holy water makes you magic immune (when it's in your system).
To counteract power-gaming and the hard-counter aspect of this change, holy water now has an OD threshold of 10u, and a metabolism rate of 1u per tick. The overdose causes screen shake, confusion and makes you say religious stuff. This is harmless outside of combat, but makes it so powergamers and validhunters will not be popping pills so easily for infinite magic resist.

## Why It's Good For The Game

The chaplain, being holy, is immune to magic. Holding (but not eating) holymelons gives immunity to magics. Clearly, in the game's fiction, the power of 'holy' gives magic immunity, and holy water is the prime way of receiving this power. This change adds value to the chaplain (and botanists) to provide holy water to the crew in the case of a wizard, or cult. It also provides a much-needed counter to magic, arguably one of the most dangerous hazards on station.

## Changelog
bugstep :cl:
balance: holy water gives players the trait antimagic
balance: adds holy water OD threshold and effect
balance: increases holy water metabolization rate
/:cl:

### CODE REVIEW

**Comment by coiax** [comment]
This is a pretty crazy power spike, in terms of fighting wizards. Holy water production, with an active chaplain, is essentially unlimited, and combined with chemistry factories, you could easily make a setup churning out large numbers of 50u holy water pills.

Holy melons expire after a certain number of "uses" blocking magic, but you could keep this production up indefinitely, and doesn't require active management, just chow down some pills.

**Comment by Incoming5643** [comment]
There is not supposed to be a counter to magic.

**Comment by brotherbeyondd** [comment]
cult are already losing 60% of rounds. This will legit kill them.

There's no need to give everyone nullrods. It's not like holy water is hard to come by, chaplains can make infinite amounts of it.

**Comment by Pugie** [comment]
FUCK NO

**Comment by joookschad** [comment]
a way to make cult lose 100% rounds, nice

**Comment by TheChosenEvilOne** [comment]
this is a terrible idea.

**Comment by D4C-420** [comment]
This will ruin wizards, screw cults harder than holy water already does, and take away one of the heretics most useful tools (Manus grasp is magic last I checked). There is also no overdose limit on holy water so nothing is stopping someone from drinking 1000 u of it. Anti-magic should stay being the chaplains niche (yes I know miners have access to 2 items with anti-magic and I think that just enables a valid hunting miner).

**Comment by CoffeeDragon16** [comment]
Yeah, mansus is magic. While I think having some more commonplace magic counter would be nice, right now there's nothing stopping you from getting the chaplain to slap a water tank and chugging a huge amount of holy water and being magic-immune for the round. Maybe this could be helped by adding an OD for holy water? Or it purges a certain percentage of holy water when magic is used?

**Comment by brotherbeyondd** [comment]
No nothing will help this coffee. Magic doesnt need more counters.

**Comment by Dorsisdwarf** [comment]
Other things than "Everything cult can do except run at you with a sword" this PR makes anyone who sips from the chaplains magic watertank immune to for up to 90 seconds:

- Magic missile
- Repulse
- Sacred Flame
- LSP
- ALL STAFF BOLTS OF ALL TYPES
- The Spellblade
- Fireball
- God Hand
- EMP
-Revenant abilities
- Timestop
- Forcewall
- Alll curse spells
- Lightning Bolt

**Comment by Pugie** [comment]
> To counteract power-gaming and the hard-counter aspect of this change, holy water now has an OD threshold of 20u. The overdose causes screen shake, confusion and makes you say religious stuff.

Alrighty. You can take 4 sips of a container of holy water and get well over a minute of antimagic immunity. This is not a significant enough low volume for OD. What you could do is increase the negative effect of holy water, but if this was significant to warrant not using holy water in large amounts then it would allow chaplain to mass produce 1000u poison tanks.

Also, lowering the volume isn't a good idea anyways, even this volume is too low. You need 25u to deconvert a cultist so this encourages more babysitting of the cultists or getting antihol or they're gonna have awful confusion when they get deconverted they won't be able to move straight.

Basically the OD isn't significant enough to warrant not using holy water in a constant way and it can never be balanced as such without removal of chaplain making tanks of holy water, which is retarded.

> The chaplain, being holy, is immune to magic. Holding (but not eating) holymelons gives immunity to magics. Clearly, in the game's fiction, the power of 'holy' gives magic immunity, and holy water is the prime way of receiving this power. 

You can try to lore in changes to make the game easier for you as security all you want, but the chaplain isn't immune to magic, his null rod is, I don't think grinding them yields holy water. This is also a not good basis to justify adding something that allows mass produced antimagic immunity.

Remember this PR? https://github.com/tgstation/tgstation/pull/50522 This PR got reverted in less than a month because cult succeeding plummeted, and  this was a less significant change than what you are opting to make. This was mindshields removing the stun ability of cult stun, what you propose is to make any and every crewmember with holy water in their systems immune to cult stun for over a minute per 4 sips of a bottle. That's outright INSANITY talk, judging by the last attempt at a cult stun nerf which removes the stun this clearly won't go well.

I don't expect you to think very hard about this PR though, this PR came out a few hours after round 153375, where you were security and got cult stunned and then ghosted, presumably out of frustration. You seem to gloss over the ramifications of your change in this PR, I'm guessing you were hoping for the slim chance to stealth in a change that would prevent you dying as security as much.

This PR has no need to be present lore wise, is godawful mechanically wise in every single way and comes out of your bizarre desire to dominate everything as sec with little resistance with no care for actually having a decent fight with stakes, I can assure you that if this PR got merged you'd have a couple rounds of cult as sec and think it's fun lawnmowering cults, then you'll realise that cult never gets past red eyes and you'll realise how shallow your victories really are, in that they mean nothing because of your incessant need to make everything biased in your favour, because you died once on one round because you got clicked first and overwhelmed by cultists because you let a cultist fight you in melee.

You could close this PR of your own volition now, or double down on your god awful attempt at making the game a single player ez mode for you as security and be remembered as the guy who died as sec and made a cult nerf pr within hours of this, as well as a maintainer closing it anyways.

You idiot.

**Comment by gorillion** [comment]
> Alrighty. You can take 4 sips of a container of holy water and get well over a minute of antimagic immunity.
Fair point - I have been trying to change the metabolization time of holy water to make it a short window (~10 seconds) of magic immunity, as I do realize that default metabolization is really slow, and 20u is still a lot. I have 0 experience with byond's programming language, and getting help is hard so I've hit a wall in that department. I pushed the update because it would be better to show *some* progress being made towards this being a proper PR.

> Also, lowering the volume isn't a good idea anyways, even this volume is too low. You need 25u to deconvert a cultist so this encourages more babysitting of the cultists or getting antihol or they're gonna have awful confusion when they get deconverted they won't be able to move straight.
> 
A deconverted cultist having under a minute of (fairly mild) confusion after they had to be strapped to a chair and fed holy water for over two minutes to be deconverted doesn't sound that much worse to me. The deconversion process is already shit, and as a result hardly ever happens, resulting in round removal for cultists. 

> I don't expect you to think very hard about this PR though, this PR came out a few hours after round 153375, where you were security and got cult stunned and then ghosted, presumably out of frustration. You seem to gloss over the ramifications of your change in this PR, I'm guessing you were hoping for the slim chance to stealth in a change that would prevent you dying as security as much.

I know you think this is a huge burn, but do you really think this handwaves the balance issues cult has because you looked at one round's logs? I have a (self imposed) cult ban, and being security, I was turned into a shard (theres some more round removal) post-mortem and forced to ghost. I've played more than my fair share of rounds against cults, both as security and mostly as non-security. It doesn't take a genius to see why an on demand 10 second stun is a balance issue.

> because you died once on one round because you got clicked first and overwhelmed by cultists because you let a cultist fight you in melee.

This is really funny. I've heard this one a few times, you just shouldn't get in melee range of a cultist. It's equivalent to "you should have just not died". Considering that shotguns were just neutered, to kill a cultist at range, a security officer, the best armed job on the station, would need to rely on an egun. An egun has 10 shots before it's empty. Cult bases are in maint rooms, very small, maybe 5x5 or 3x3, with 1 tile wide hallways leading up to them. A cultist wearing robes takes 10 energy shots just to knock down (not kill, mind you!). Without robes that's 6 shots. But of course, just don't let them fight you in melee. Cultists are antags, and yes, they should have power to do their antagonizing. But their balance is nonexistent. What's worse, a 10-second stun with one click, or having to overpower and kill someone using the actual environmental combat that is staple to SS13? You know, like *another* team based antag has to do, and seems to do pretty well at?
Don't pretend.

**Comment by Pugie** [comment]
The fact you have a self imposed cult ban makes the fact you're making such heavy handed crew sided cult nerfs all the more insulting and deranged.
I have nothing more to say in this PR, you can try connect the dots to why this PR sucks yourself.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 1]
```diff
@@ -269,14 +273,26 @@
...
-	if(data["misc"] >= 60)	// 30 units, 135 seconds
+	if(data["misc"] >= 30)	// 30 units, 54 seconds
 		if(iscultist(M))
 			SSticker.mode.remove_cultist(M.mind, FALSE, TRUE)
 		M.jitteriness = 0
 		M.stuttering = 0
 		holder.remove_reagent(type, volume)	// maybe this is a little too perfect and a max() cap on the statuses would be better??
 		return
-	holder.remove_reagent(type, 0.4)	//fixed consumption to prevent balancing going out of whack
+	holder.remove_reagent(type, metabolization_rate)	//this line sucked and it is now fixed by Bug
```
Comment (Jared-Fogle) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:283]:
```suggestion
	holder.remove_reagent(type, metabolization_rate)
```

don't need comments explaining what code used to be

**Comment by 81Denton** [comment]
I have nothing to add to pugie's comment: https://github.com/tgstation/tgstation/pull/55896#issuecomment-753526003

**Comment by ExcessiveUseOfCobblestone** [comment]
please don't use PR comments to attack the creator (aka call them an idiot), just the PR :)

**Pull request closed without merging**